templates/plugin/container_gateway.yml.erb

```diff
@@ -6,3 +6,17 @@
 <% if scope.lookupvar("foreman_proxy::plugin::container_gateway::sqlite_timeout") -%>
 :sqlite_timeout: <%= scope.lookupvar("foreman_proxy::plugin::container_gateway::sqlite_timeout") %>
 <% end -%>
+<%-
+case scope.lookupvar('foreman_proxy::plugin::container_gateway::database_backend')
+when 'postgres'
+  uri = URI("postgres://")
+  uri.user = scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_user')
+  uri.password = scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_password')
+  uri.host = scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_host')
+  uri.port = scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_port')
+  uri.path = "/#{scope.lookupvar('foreman_proxy::plugin::container_gateway::postgresql_database')}"
+when 'sqlite'
+  uri = URI("sqlite://#{scope.lookupvar("foreman_proxy::plugin::container_gateway::sqlite_db_path")}")
```

Comment:
It's weird that `sqlite://$path` ends up being simplified to `sqlite:$path`. Is that really correct? Perhaps the Ruby URI class isn't entirely correct?

Reply:
yeah I'm working through this myself. sqlite:/potato.db does indeed work, just tested with Sequel. I don't really like having to work around that though :P

Reply:
I just pushed -- I removed the URI usage for sqlite.